integration/tests/upgrades/suite.go

```diff
@@ -0,0 +1,179 @@
+package upgrades
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/engine/ghost/client"
+	"github.com/onflow/flow-go/integration/testnet"
+	"github.com/onflow/flow-go/integration/tests/lib"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type Suite struct {
+	suite.Suite
+	log zerolog.Logger
+	lib.TestnetStateTracker
+	cancel  context.CancelFunc
+	net     *testnet.FlowNetwork
+	ghostID flow.Identifier
+	exe1ID  flow.Identifier
+}
+
+func (s *Suite) Ghost() *client.GhostClient {
+	ghost := s.net.ContainerByID(s.ghostID)
+	client, err := lib.GetGhostClient(ghost)
+	require.NoError(s.T(), err, "could not get ghost client")
+	return client
+}
+
+func (s *Suite) AccessClient() *testnet.Client {
+	chain := s.net.Root().Header.ChainID.Chain()
+	client, err := testnet.NewClient(fmt.Sprintf(":%s", s.net.AccessPorts[testnet.AccessNodeAPIPort]), chain)
+	require.NoError(s.T(), err, "could not get access client")
+	return client
+}
+
+func (s *Suite) ExecutionClient() *testnet.Client {
+	execNode := s.net.ContainerByID(s.exe1ID)
+	chain := s.net.Root().Header.ChainID.Chain()
+	client, err := testnet.NewClient(fmt.Sprintf(":%s", execNode.Ports[testnet.ExeNodeAPIPort]), chain)
+	require.NoError(s.T(), err, "could not get execution client")
+	return client
+}
+
+type AdminCommandRequest struct {
```

Comment:
Peter has added a utility for interacting with the admin server [here](https://github.com/onflow/flow-go/pull/4121/files#diff-1ec0399238f585fe8a9299a71d1a9225513e654c57979aaf145e3839e56be5d6R1-R108) -- we could pull it in here to avoid duplication.

Reply:
Done.